XcodeServerSDK/XcodeServer.swift

```diff
@@ -472,6 +530,88 @@ public extension XcodeServer {
+                completion(repositories: nil, error: error)
+                return
+            }
+            
+            guard let repositoriesBody = (body as? NSDictionary)?["results"] as? NSArray else {
+                completion(repositories: nil, error: Error.withInfo("Wrong body \(body)"))
+                return
+            }
+            
+            let repos: [Repository] = XcodeServerArray(repositoriesBody)
+            completion(repositories: repos, error: nil)
+        }
+    }
+    
+    /**
+    Enum with response from creation of repository.
+    
+    - RepositoryAlreadyExists: Repository with this name already exists on OS X Server.
+    - NilResponse:             Self explanatory.
+    - CorruptedJSON:           JSON you've used to create repository.
+    - WrongStatusCode:         Something wrong with HHTP status.
+    - Error:                   There was an error during netwotk activity.
+    - Success:                 Repository was successfully created 🎉
+    */
+    public enum CreateRepositoryResponse {
+        case RepositoryAlreadyExists
+        case NilResponse
+        case CorruptedJSON
+        case WrongStatusCode(Int)
+        case Error(ErrorType)
+        case Success(Repository)
+    }
+    
+    /**
+    XCS API call for creating new repository on configured Xcode Server.
+    
+    - parameter repository: Repository object.
+    - parameter repository: Optional object of created repository.
+    - parameter error:      Optional error.
+    */
+    public func createRepository(repository: Repository, completion: (repsponse: CreateRepositoryResponse) -> ()) {
+        let body = repository.dictionarify()
+        
+        self.sendRequestWithMethod(.POST, endpoint: .Repositories, params: nil, query: nil, body: body) { (response, body, error) -> () in
+            if let error = error {
+                completion(repsponse: XcodeServer.CreateRepositoryResponse.Error(error))
+                return
+            }
+            
+            guard let response = response else {
+                completion(repsponse: XcodeServer.CreateRepositoryResponse.NilResponse)
+                return
+            }
+            
+            guard let repositoryBody = body as? NSDictionary where response.statusCode == 204 else {
+                switch response.statusCode {
+                case 200:
+                    completion(repsponse: XcodeServer.CreateRepositoryResponse.CorruptedJSON)
+                case 409:
+                    completion(repsponse: XcodeServer.CreateRepositoryResponse.RepositoryAlreadyExists)
+                default:
+                    completion(repsponse: XcodeServer.CreateRepositoryResponse.WrongStatusCode(response.statusCode))
+                }
+
+                return
+            }
+            
+            let repository = Repository(json: repositoryBody)
+            completion(repsponse: XcodeServer.CreateRepositoryResponse.Success(repository))
```

Comment:
Nice code design here, I really like how you guard for the status code being `204`, otherwise you go into investigating what exactly went wrong. :+1: